pkg/cmd/pulumi/errors.go

```diff
@@ -104,47 +104,73 @@ func printDecryptError(e engine.DecryptError) {
 // prints out a panic-like banner with information about the error and how to
 // report it.
 func printSnapshotIntegrityError(err error, sie deploy.SnapshotIntegrityError) {
-	readOrWrite := ""
-	if sie.Op == deploy.SnapshotIntegrityRead {
-		readOrWrite = `
+	var message strings.Builder
+	message.WriteString(`The Pulumi CLI encountered a snapshot integrity error. This is a bug!
+
+================================================================================
+We would appreciate a report: https://github.com/pulumi/pulumi/issues/
+`)
+
+	if sie.Op == deploy.SnapshotIntegrityRead && sie.Metadata == nil {
+		message.WriteString(`
 NOTE: This error occurred while reading a snaphot. This error was introduced by
 a previous operation when it wrote the snapshot. If you have details about that
 operation, please include them in your report as well.
-`
+`)
 	}
 
-	cmdutil.Diag().Errorf(diag.RawMessage(
-		"", /*urn*/
-		fmt.Sprintf(`The Pulumi CLI encountered a snapshot integrity error. This is a bug!
-
-================================================================================
-We would appreciate a report: https://github.com/pulumi/pulumi/issues/
-%s
+	message.WriteString(fmt.Sprintf(`
 Please provide all of the text below in your report.
 ================================================================================
-Pulumi Version:    %s
 Go Version:        %s
 Go Compiler:       %s
 Architecture:      %s
-Operating System:  %s
+Operating System:  %s`,
+		runtime.Version(),
+		runtime.Compiler,
+		runtime.GOARCH,
+		runtime.GOOS,
+	))
+
+	if sie.Op == deploy.SnapshotIntegrityRead && sie.Metadata != nil {
+		message.WriteString(fmt.Sprintf(`
+
+Write Version:     %s
```

Comment:
If the versions match here we could potentially collapse them, but I don't think it matters too much one way or another.